negotiation/neural/trainer.py

```diff
@@ -235,7 +235,13 @@ def validate(self, valid_iter):
             targets = batch.targets
             lengths = batch.lengths
 
-            outputs, attns, _ = self.model(encoder_inputs, decoder_inputs, lengths)
+            item_title = batch.item_title
+            previous_turns = batch.prev_turns
+            combined_inputs = {'enc': encoder_inputs,
+                              'item': item_title,
+                              'prev': previous_turns}
+
+            outputs, attns, _ = self.model(combined_inputs, decoder_inputs, lengths)
```

Comment:
combined inputs:
separate encoder inputs from context inputs, so you can have encoder_inputs, decoder_inputs, context_inputs, length